simpleocr/files.py

```diff
@@ -45,45 +32,152 @@ def write(self):
+        - Relative path with or without extension
+        - File in DATA_DIRECTORY with or without extension
+        - Absolute path with or without extension
+        :param image: image path in str type
+        :return: The absolute image path
+        """
+        # If the path exists, return the path, but make sure it's an absolute path first
+        if os.path.exists(image):
+            return os.path.abspath(image)
+        # Try to find the file with the passed path with the various extensions
+        image_with_extension = try_extensions(IMAGE_EXTENSIONS, os.path.splitext(image)[0])
+        if image_with_extension:
+            return os.path.abspath(image_with_extension)
+        # The file must be in the data directory if it has not yet been found
+        image_basename = os.path.basename(image)
+        image_datadir = try_extensions(IMAGE_EXTENSIONS, os.path.join(DATA_DIRECTORY, image_basename))
+        if image_datadir:
+            return os.path.abspath(image_datadir)
+        # The file cannot be found, so raise a FileNotFound Error
+        raise Image.FileNotFound
+
+    @staticmethod
+    def get_array_from_pil(pillow):
+        """
+        Returns an OpenCV compatible array from a Pillow Image object
+        :param pillow: pillow image object
+        :return: OpenCV compatible numpy array
+        """
+        imagefile = numpy.array(pillow)
+        return imagefile[:, :, ::-1].copy()
 
     @property
     def is_grounded(self):
-        """checks if this file is grounded"""
+        """
+        :return: True if Image is grounded
+        """
         return not (self.ground is None)
 
-    def set_ground(self, segments, classes, write_file=False):
-        """creates the ground, saves it to a file"""
-        if self.is_grounded:
+    @property
+    def has_file_reference(self):
+        """
+        :return: True if Image has a file reference
+        """
+        return self.path is not None
+
+    def set_ground(self, segments, classes, write=False):
+        """
+        Creates the ground data in memory, optionally writing it to a ground file,
+        if write is set to True
+        :param segments: segments to be grounded
+        :param classes: classes to be grounded
+        :param write: if True, writes the ground data to a ground file
+        :return: None
+        """
+        if self.is_grounded and self._debug:
             print("Warning: grounding already grounded file")
         self.ground = GroundFile(self.ground_path)
         self.ground.segments = segments
         self.ground.classes = classes
-        if write_file:
+        if write:
+            if not self.ground_path:
+                raise ValueError("Cannot write ground file for an Image without file reference")
             self.ground.write()
 
-    def remove_ground(self, remove_file=False):
-        """removes ground, optionally deleting it's file"""
+    def remove_ground(self, remove=False):
```

Comment:
Having a function named `remove_ground` with a boolean parameter called `remove` is extremely confusing.
What's wrong with `remove_file` and `write_file`?